#acl TonyZuccarino:read,write,delete
## backup of page "Known Issues" submitted 2005-09-20 13:17:01
 * make sure you can't login using the encrypted password (with {SHA} in front..)..
      * make sure the email-password thing (old) doesn't work and the (new) system does work.
      * need better logout thinger?
 * intuitive redirect labels (wikipedia)

= bugs = 
 * thumbnail macro doesn't have proper regexp for image captions with comas in them
 * thumnail macro also needs to work perfectly in alignment with the syntax for {{{attachment:a%20%image%20with%20spaces__and__these__things}}}
   * a particular one that it doesn't work with is {{{C__Documents%20and%20Settings_Leviathan_tercero.JPG}}}
   * works with that image referenced ''as'' {{{C__Documents and Settings_Leviathan_tercero.JPG}}} -- isn't doing %20 stuff?
 * Uploading from IE on windows can produce said ugly attachments. EW.
 * Sub-pages.  In general need to fix simpleParse.
 * Interwiki.
 * User Statistics on www.bearvalleywiki.org just doesn't want to update with pages that I have created, nor the time of last edit. Go to my wiki and you'll see the User Stat is definitely behind what Recent Changes would indicate. Mike Ivanov looked at it for a while. Didn't see anything obvious. 
    I would check to see if the file userstats.xml has the correct permissions.  It's in the util_apps directory.  You can just recreate it by making a text file containing:  {{{
<?xml version="1.0" ?>
<stats></stats>
}}}
    I won't have regularly net access until the 6th, so I can't help beyond that, sorry!
             Hi Philip, userstats.xml is 644, same owner.group as the rest of the files. BTW, when I run 'python ./rebuildlastedited.py' in the LocalWiki directory, it gets into an infinite loop or blocks waiting for something. Trying to look into it a bit.
                   I think found a problem. The clue was in the Apache error log where Python was putting its errors... one constant error was that somewhere in the add_to_userstats (in util_apps) call stack a python dom module trying to open userstats.xml as '-7/userstats.xml'. Looking at add_to_userstats code, I see that that the userStatAdd function expects four arguments, and PageEditor.py below was calling userStatAdd with five, and looks like argv[1] was the time offset, which, indeed is -7. The remaining four args sounded right, but argv[1] seemed extra. I deleted the str(config.tx_offset) and now User Statistics *seems* to be working. Below is the code line in PageEditor I commented out and what I put in. 

                      OLD # pid = os.spawnl(os.P_WAIT, config.app_dir + '/add_to_userstats', config.app_dir + '/add_to_userstats', str(config.tz_offset), config.app_dir, self.request.user.name, action, self.page_name)

                      NEW pid = os.spawnl(os.P_WAIT, config.app_dir + '/add_to_userstats', config.app_dir + '/add_to_userstats', config.app_dir, self.request.user.name, action, self.page_name)

                         Update: MikeIvanov found that I had an old ver of add_to_userstats. The latest ver of that file was consistent with the calling structure of the OLD statement in PageEditor

= more general =
 * set up subversion completely.  document the code setup!

= speed fixes =
 * ["Title Index"], etc should be fast or turned off.  Do per-day generation of Wanted Pages, Orphaned Pages, and Outgoing Links.
